internal/controller/nonadminrestore_controller.go

```diff
@@ -0,0 +1,418 @@
+			Message: "restore accepted",
+		},
+	)
+	if updated {
+		if err := r.Status().Update(ctx, nar); err != nil {
+			logger.Error(err, nonAdminRestoreStatusUpdateFailureMessage)
+			return false, err
+		}
+		logger.V(1).Info("NonAdminRestore condition set to Accepted")
+	}
+	return false, nil
+}
+
+func (r *NonAdminRestoreReconciler) setUUID(ctx context.Context, logger logr.Logger, nar *nacv1alpha1.NonAdminRestore) (bool, error) {
+	// Get the latest version of the NAR object just before checking if the NACUUID is set
+	// to ensure we do not miss any updates to the NAR object
+	narOriginal := nar.DeepCopy()
+	if err := r.Get(ctx, types.NamespacedName{Name: narOriginal.Name, Namespace: narOriginal.Namespace}, nar); err != nil {
+		logger.Error(err, "Failed to re-fetch NonAdminRestore")
+		return false, err
+	}
+
+	if nar.Status.VeleroRestore == nil || nar.Status.VeleroRestore.NACUUID == constant.EmptyString {
+		veleroRestoreNACUUID := function.GenerateNacObjectUUID(nar.Namespace, nar.Name)
+		nar.Status.VeleroRestore = &nacv1alpha1.VeleroRestore{
+			NACUUID:   veleroRestoreNACUUID,
+			Namespace: r.OADPNamespace,
+			Name:      veleroRestoreNACUUID,
+		}
+		if err := r.Status().Update(ctx, nar); err != nil {
+			logger.Error(err, nonAdminRestoreStatusUpdateFailureMessage)
+			return false, err
+		}
+		logger.V(1).Info(veleroRestoreReferenceUpdated)
+	} else {
+		logger.V(1).Info("NonAdminRestore already contains VeleroRestore UUID reference")
+	}
+	return false, nil
+}
+
+func (r *NonAdminRestoreReconciler) setFinalizer(ctx context.Context, logger logr.Logger, nar *nacv1alpha1.NonAdminRestore) (bool, error) {
+	added := controllerutil.AddFinalizer(nar, constant.NonAdminRestoreFinalizerName)
+	if added {
+		if err := r.Update(ctx, nar); err != nil {
+			logger.Error(err, "Failed to add finalizer")
+			return false, err
+		}
+		logger.V(1).Info("Finalizer added to NonAdminRestore")
+	} else {
+		logger.V(1).Info("Finalizer already exists on NonAdminRestore")
+	}
+	return false, nil
+}
+
+func (r *NonAdminRestoreReconciler) createVeleroRestore(ctx context.Context, logger logr.Logger, nar *nacv1alpha1.NonAdminRestore) (bool, error) {
+	if nar.Status.VeleroRestore == nil || nar.Status.VeleroRestore.NACUUID == constant.EmptyString {
+		return false, errors.New("unable to get Velero Restore UUID from NonAdminRestore Status")
+	}
+
+	veleroRestoreNACUUID := nar.Status.VeleroRestore.NACUUID
+
+	veleroRestore, err := function.GetVeleroRestoreByLabel(ctx, r.Client, r.OADPNamespace, veleroRestoreNACUUID)
+
+	if err != nil {
+		// Case in which more then one VeleroBackup is found with the same label UUID
+		logger.Error(err, findSingleVRError, uuidString, veleroRestoreNACUUID)
+		return false, err
+	}
+
+	if veleroRestore == nil {
+		logger.Info("VeleroRestore with label not found, creating one", uuidString, veleroRestoreNACUUID)
+		restoreSpec := nar.Spec.RestoreSpec.DeepCopy()
+		restoreSpec.IncludedNamespaces = []string{nar.Namespace}
```

Comment:
this is broken

velero restore spec must contain velero backup name, copying from non admin restore will have non admin backup name

Reply:
fixed.